walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,18 +62,46 @@
     }
     
     List<Event> mandatoryAttendeeEvents = new ArrayList<Event>();
+    List<Event> optionalAttendeeEvents = new ArrayList<Event>();
     for (Event event : allEvents) {
       if (!optionalAttendeesOnly(event, mandatoryRequestAttendees)) {
         mandatoryAttendeeEvents.add(event);
+      } else {
+        optionalAttendeeEvents.add(event);
       }
     }
     sortByStart(mandatoryAttendeeEvents);
+    sortByStart(optionalAttendeeEvents);
     
-    if (!mandatoryAttendeeEvents.isEmpty()) {
-      return getAllPossibleTimes(mandatoryAttendeeEvents, request);
+    if (mandatoryAttendeeEvents.isEmpty()) {
+      return everyoneTimeOptions;
+    }
+    
+    ArrayList<TimeRange> mandatoryAttendeeTimeOptions = getAllPossibleTimes(mandatoryAttendeeEvents, request);
+    
+    List<ArrayList<TimeRange>> possibleTimeOptionsCollection = new ArrayList<ArrayList<TimeRange>>();
+    possibleTimeOptionsCollection.add(mandatoryAttendeeTimeOptions);
+    
+    List<ArrayList<Event>> optionalAttendeeEventsCollection = new ArrayList<ArrayList<Event>>();
+    
+    // Sorts events according to their optional attendees, making a separate ArrayList for each optional
+    // attendee's events.
+    for (String optionalAttendee : optionalRequestAttendees) {
+      ArrayList<Event> thisOptionalAttendeeEvents = new ArrayList<Event>();
+      for (Event event : optionalAttendeeEvents) {
+        if (event.getAttendees().contains(optionalAttendee)) {
+          thisOptionalAttendeeEvents.add(event);
+        }
+      }
+      sortByStart(thisOptionalAttendeeEvents);
```

Comment:
If you sort `optionalAttendeeEvents` before breaking it down by optional attendee, you won't have to do this sort because all events will already be sorted by start time.

Reply:
Done